### ISSUE TITLE
Duration.fromObject({ minutes: 30 }).toFormat('HH:mm') returns HH:30

### ISSUE THREAD

**Issue by reinaldorauch** [moment/luxon#288]
The documentation implies that the Duration.toFormat uses the same tokens as DateTime.toFormat. Is that correct?

When I use hh:mm in format, it outputs 00:30, which seems correct, but if the duration is `{hours: 1231, minutes: 30}` the output is 1231:30 witch is correct but the behavior differs from the [table](https://moment.github.io/luxon/docs/manual/formatting.html#table-of-tokens)

Thanks for your attention.

**Comment by icambron** [comment]
> The documentation implies that the Duration.toFormat uses the same tokens as DateTime.toFormat. Is that correct?

Not really. Only a subset of the DateTime tokens apply to Durations (e.g. "time zone" and "24-hour time" doesn't mean anything for Durations). Additionally, as you point out, the must be interpreted somewhat differently for Durations. It's its own animal, and it should be documented accordingly. I'd love to create a whole separate section in the formatting page with its own table.

**Comment by networkimprov** [comment]
Could you summarize the Duration#toFormat() tokens here?

Also can toFormat() provide the total count of days (i.e. not total % 365)? Token 'o' doesn't work, tho it seemed like the right concept. If there isn't a token for it, how can I derive it?

**Comment by icambron** [comment]
Does this help?

```js
const tokenToField = token => {
        switch (token[0]) {
          case 'S':
            return 'millisecond';
          case 's':
            return 'second';
          case 'm':
            return 'minute';
          case 'h':
            return 'hour';
          case 'd':
            return 'day';
          case 'M':
            return 'month';
          case 'y':
            return 'year';
          default:
            return null;
        }
      },
```

I see what you mean about the aliases, but we could also just document them.

**Comment by networkimprov** [comment]
Also document that most (if not all, which?) of these report a total if no larger unit appears in the format, e.g. for `Duration.fromMillis(366*24*60*60*1000)`

`"d"` => 366  
`"y d"` => 1 1

**Comment by icambron** [comment]
They always do. It calls `duration.shiftTo({ whatever units are in format })`

**Comment by networkimprov** [comment]
And add that token repetition yields padding, e.g. for `var d = Duration.fromMillis(366*24*60*60*1000)`

`d.toFormat("yy dd hh mm ss")` => 01 01 00 00 00

How can a duration have a meaningful number of months? Do you assume that it starts on Jan 1?

Does month & year calculation include Feb 29 every four years, and if so in which of the four?

**Comment by icambron** [comment]
@networkimprov depends: https://moment.github.io/luxon/docs/manual/math.html#casual-vs-longterm-conversion-accuracy

**Comment by icambron** [comment]
Fixed in 421b3c8a41a5b9bc4d16dcc2df212096e8ea77b5

**Comment by networkimprov** [comment]
I don't see mention of padding, e.g. `d.toFormat("yy dd hh mm ss")` in the patch...?

**Comment by icambron** [comment]
@networkimprov yes, should add the padded ones in there too
